### PR TITLE
doc: kernel: workqueue: various updates

### PR DESCRIPTION
**Pull request by pabigot**
General cleanup, particularly in precise usage of terminology like scheduled, queued, running, and pending.  Integration of the Best Practices section from #33104.  A tweak that may help with the concerns of #33761.

### CODE REVIEW

File: doc/reference/kernel/threads/workqueue.rst [hunk 1]
```diff
@@ -233,7 +233,7 @@ rescheduling can be controlled by the optional final parameter; see
 The following API can be used to interact with a workqueue:
 
 * :c:func:`k_work_queue_drain()` can be used to block the caller until no more
```
Comment (jfischer-no) [inline doc/reference/kernel/threads/workqueue.rst:251]: until ~~no more~~
the work queue has no items left. ?

**Comment by nashif** [comment]
you need to rebase

File: doc/reference/kernel/threads/workqueue.rst [hunk 2]
```diff
@@ -370,6 +397,132 @@ synchronization objects.  These objects should not be allocated on a stack if
...
+state that's being accessed by another thread or interrupt.  Such state might
+be a flag indicating that work needs to be done, or a shared object that is
+filled by an ISR or thread and read by the work handler.
+
+For simple flags :ref:`atomic_v2` may be sufficient.  In other cases spin
+locks (:c:struct:`k_spinlock_t`) or thread-aware locks (:c:struct:`k_sem`,
+:c:struct:`k_mutex` , ...) may be used to ensure data races don't occur.
+
+If the selected lock mechanism can :ref:`api_term_sleep` then allowing the
+work thread to sleep will starve other work queue threads, which may need to
```
Comment (mniestroj) [inline doc/reference/kernel/threads/workqueue.rst:419]:
```suggestion
work thread to sleep will starve other work queue items, which may need to
```
?

File: doc/reference/kernel/threads/workqueue.rst [hunk 3]
```diff
@@ -247,8 +248,8 @@ rescheduling can be controlled by the optional final parameter; see
 
 The following API can be used to interact with a workqueue:
 
-* :c:func:`k_work_queue_drain()` can be used to block the caller until no more
-  the work queue has no items left.  Work items resubmitted from the workqueue
+* :c:func:`k_work_queue_drain()` can be used to block the caller until the
+  work queue has no items left.  Work items resubmitted from the workqueue
```
[the remark this answers is not in the document]
Comment (mniestroj) [reply doc/reference/kernel/threads/workqueue.rst]: This comment is still valid, changes should go into "doc: kernel: workqueue: correct terminology misuse".

[the remark this answers is not in the document]
Comment (pabigot) [reply doc/reference/kernel/threads/workqueue.rst]: moved

**Review by mniestroj — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM, best practices section is very much appreciated.

**Pull request merged**